gdal/frmts/wms/gdalwmscache.cpp

```diff
@@ -31,89 +33,269 @@
 
 CPL_CVSID("$Id$")
 
-GDALWMSCache::GDALWMSCache() :
-    m_cache_path("./gdalwmscache"),
-    // No need to do the default: m_postfix("");
-    m_cache_depth(2)
-{}
 
-GDALWMSCache::~GDALWMSCache() {}
+static void CleanCacheThread( void *pData )
+{
+    GDALWMSCache *pCache = static_cast<GDALWMSCache *>(pData);
+    pCache->Clean();
+}
 
-CPLErr GDALWMSCache::Initialize(CPLXMLNode *config) {
-    const char *xmlcache_path = CPLGetXMLValue(config, "Path", nullptr);
-    const char *usercache_path = CPLGetConfigOption("GDAL_DEFAULT_WMS_CACHE_PATH", nullptr);
-    if(xmlcache_path)
+//------------------------------------------------------------------------------
+// FileCache
+//------------------------------------------------------------------------------
+class FileCache : public GDALWMSCacheImpl
```

Comment:
can you prefix the class name ? FileCache has a risk of clashing with code external to GDAL

Reply:
Fixed